### PR TITLE
Implicit conversion

### PR DESCRIPTION
**Pull request by micooke**
add explicit variable type conversions to address MSVC C4267,C4244 warnings

### CODE REVIEW

File: test/bench.c [hunk 1]
```diff
@@ -44,4 +44,4 @@ int main() {
 
     const EmlError e = bench_all();
     return -(int)e;
-}
+}
```
Comment (micooke) [inline test/bench.c]: whoops

**Comment by jonnor** [comment]
Uncovered suprisingly many implicit conversion warnings. Why doesn't Linux/GCC version pick this up? Is there maybe a compiler flag we could/should add?

**Comment by jonnor** [comment]
Anyway, looks good!

**Comment by micooke** [comment]
\_(ツ)_/¯
Some were picked up running ```test_benchmark.py``` as it errored out. The others i picked up in the stdout when running ```setup.py```

**Comment by jonnor** [comment]
Ah OK, so they are warnings under GCC as well. I guess setup.py should stay *without* `-Werror` in general, as people run that on their unknown compiler versions all the time, and warnings breaking the build would be super annoying. But maybe there is a way to enable it only on Travis CI for example

**Comment by jonnor** [comment]
Oh, seems we already have that. But GCC version is a bit old, so I guess the enabled warnings are just a bit more conservative there.

**Pull request merged**